source/Halibut/HalibutRuntime.cs

```diff
@@ -25,7 +25,7 @@ public class HalibutRuntime : IHalibutRuntime
         readonly ConcurrentDictionary<Uri, IPendingRequestQueue> queues = new();
         readonly IPendingRequestQueueFactory queueFactory;
         readonly X509Certificate2 serverCertificate;
-        readonly List<IDisposable> listeners = new();
+        readonly MutexedItem<List<IAsyncDisposable>> listeners = new MutexedItem<List<IAsyncDisposable>>(new List<IAsyncDisposable>());
```

Comment:
Just to verify, the motivation behind this change was to make disposal async, right?

Reply:
We needed to make accept be async (with a cancellation token)
Which meant that to finish the accept loop task we needed to await it which meant a IAsyncDisposable.
To do that meant we needed this to be a list of `IAsyncDisposable`
To do that and to preserve the only one thing can access this list at a time behaviour we needed some way of locking in both async and sync context (`lock` does not work with async)
To ensure that is done properly this type ensures, assuming you don't try really hard, that you can't have access to the list unless it is mutually exclusive access.